graphql/writer.go

```diff
@@ -0,0 +1,109 @@
+package graphql
+
+import (
+	"encoding/json"
+	"sync"
+)
+
+// errorRecorder is a concurrency-safe way where we can record the first error
+// we get from executing the graphql query.
+type errorRecorder struct {
+	sync.Once
+	err error
+}
+
+func (e *errorRecorder) record(err error) {
+	if err == nil {
```

Comment:
i don't think this is safe as record() might be invoked in another goroutine and setting error pointer

let's get rid of this line?

Reply:
this checks the passed in error, should be safe I think?

Reply:
yes i misread